src/pages/BuyDomain/BuyDomain.tsx

```diff
@@ -0,0 +1,70 @@
+import type { FC } from 'react';
+
+import { useState } from 'react';
+import {
+	IconZero,
+	DomainSearch,
+	BuyDomainModal,
+	ConnectWallet,
+} from '../../components';
+import { useWeb3 } from '../../hooks';
+import styles from './BuyDomain.module.scss';
+
+export const BuyDomain: FC = () => {
+	const { account } = useWeb3();
+
+	const [buyDomainModal, setBuyDomainModal] = useState<{
+		isOpen;
+		domainName;
+	}>({
+		isOpen: false,
+		domainName: '',
+	});
+
+	const handleOnBuyButtonClick = (domainName) => {
+		setBuyDomainModal({
+			isOpen: true,
+			domainName,
+		});
+	};
+
+	const handleOnModalClose = () => {
+		setBuyDomainModal({
+			...buyDomainModal,
+			isOpen: false,
+		});
+	};
+
+	if (!account) {
+		return (
+			<div className={styles.Container}>
+				<div className={styles.Wrapper}>
+					<ConnectWallet
+						message={'Connect a Web3 wallet to see your Buy Domains data.'}
+					/>
+				</div>
+			</div>
+		);
+	}
+
+	return (
+		<>
+			<div className={styles.Container}>
+				<div className={styles.Wrapper}>
+					<div className={styles.Header}>
+						<IconZero />
+						<h1>Buy Your Domain</h1>
+					</div>
+
+					<div className={styles.Content}>
+						<DomainSearch onBuyButtonClick={handleOnBuyButtonClick} />
+					</div>
+				</div>
+			</div>
+
+			{buyDomainModal.isOpen && <BuyDomainModal onClose={handleOnModalClose} />}
```

Comment:
It would be good to follow Radix's UI pattern, which is to remove conditional rendering and use the `open` prop on the Modal to control rendering. Happy for discussion here though!

Reply:
Updated.